src/v/config/leaders_preference.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/leaders_preference.h"
+
+#include <boost/algorithm/string.hpp>
+
+namespace config {
+
+std::ostream& operator<<(std::ostream& os, const leaders_preference& lp) {
```

Comment:
nit: we plan to get rid of the `operator<<` and replace them with `fmt::formatter` can you change that to the formatter ?

Reply:
Tried this, but looks like we still need `operator<<` [here](https://github.com/redpanda-data/redpanda/blob/dev/src/v/config/property.h#L146), and implementing the formatter doesn't automatically get you an `operator<<`.

Reply:
ha, good point